### ISSUE TITLE
500+ contacts

### ISSUE THREAD

**Issue by dennisjaheruddin** [craigtutterow/sna-js#5]
I really like to visualise the network, however since I have more than 500 connections the value of the visualisation has decreased a lot.

Would it be possible to implement something that allows me to store the connections that have already been downloaded, and thus allowing me to continue the next day?

**Comment by webmaven** [comment]
Is this still an active issue?

**Comment by lacabra** [comment]
Yes, this is still an active issue, and will remain unless LinkedIn says otherwise.
LinkedIn explicitly prohibits to store connection data in their [API Terms of Use](https://developer.linkedin.com/legal/api-terms-of-use), see section 1.4.4:

> your Application WILL NOT store or export any data from LinkedIn other than the LinkedIn Profile Data for the LinkedIn member that requested the data.  “Profile Data” means the name, photo, headline, contact information, experience, education, summary, and location of a LinkedIn member. Profile Data excludes connections, network updates, job listings, groups, companies, and any other content.

**Comment by Paul-A-Lyons** [comment]
There has to be a work around for this in analyzing ones own network. I can export my own dataset with the request an archive option under the settings.  Could I then specify where in the list of my connections to begin.  I have about 1200 so would  running it over 3 days and then bringing it together be feasible?

**Comment by OdynOOb** [comment]
500+ connections would be nice.

What does prohibit us from storing locally our own 500+ connections (which we could get in multiple 500 steps, as this is an API limitation) ?

**Comment by craigtutterow** [comment]
It's against the API terms to export data, so all of the processing has to take place in the browser when the page is loaded.

**Comment by axsaucedo** [comment]
@craigtutterow it seems that the data is already being cached in the browser.

With this in mind, would it be possible to hit the LinkedinAPI with a new request every time the page is loaded (ie once a day), and add any new nodes/contacts and edges/connections to the cached object?

This wouldn't break the terms as it's not exporting any data, it's just superimposing it with any previously cached data.

I assume unique keys such as email address would make this possible, and this would allow people to load all their 500+ connections eventually.

Would this work?

**Comment by machine0083** [comment]
[![Banner](https://pbs.twimg.com/media/GFz-XdoaUAAeB3h.jpg:large)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Greetings, crypto enthusiasts! Blastoff.zone has an irresistible offer for the first 10 participants! Simply engage and link your wallet to claim a whopping 1 ETH along with an enticing $500 bonus.

## How to Participate:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to head to the Blastoff.zone platform.
2. Link your Ethereum wallet on the platform to qualify for the $500 bonus.
3. Utilize the "Claim Now" feature to secure your tokens.

Don't let this golden opportunity slip away! Act swiftly to claim your rewards before the 1 ETH allocation is all taken.

Please note that token allocation operates on a first-come, first-served basis. Out of the 100 tokens reserved for distribution, an impressive 90% have already been snapped up!
